op-batcher/batcher/driver.go

```diff
@@ -492,6 +476,77 @@ func (l *BatchSubmitter) loop() {
 	}
 }
 
+func (l *BatchSubmitter) processReceiptsLoop(receiptsCh chan txmgr.TxReceipt[txRef], receiptsLoopDone chan struct{}) {
+	l.Log.Info("Starting receipts processing loop")
+	for {
+		select {
+		case r := <-receiptsCh:
+			l.txpoolMutex.Lock()
+			if errors.Is(r.Err, txpool.ErrAlreadyReserved) && l.txpoolState == TxpoolGood {
+				l.txpoolState = TxpoolBlocked
+				l.txpoolBlockedBlob = r.ID.isBlob
+				l.Log.Info("incompatible tx in txpool", "is_blob", r.ID.isBlob)
+			} else if r.ID.isCancel && l.txpoolState == TxpoolCancelPending {
+				// Set state to TxpoolGood even if the cancellation transaction ended in error
+				// since the stuck transaction could have cleared while we were waiting.
+				l.txpoolState = TxpoolGood
+				l.Log.Info("txpool may no longer be blocked", "err", r.Err)
+			}
+			l.txpoolMutex.Unlock()
+			l.Log.Info("Handling receipt", "id", r.ID)
+			l.handleReceipt(r)
+		case <-receiptsLoopDone:
+			l.Log.Info("Receipts processing loop done")
+			return
+		}
+	}
+}
+
+// throttlingLoop monitors the backlog in bytes we need to make available, and throttles incoming
+// data appropriately to keep it under a threshold.
+func (l *BatchSubmitter) throttlingLoop(throttlingLoopDone chan struct{}) {
+	l.Log.Info("Starting DA throttling loop")
+	ticker := time.NewTicker(l.Config.ThrottleInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			ctx, cancel := context.WithTimeout(l.shutdownCtx, l.Config.NetworkTimeout)
+			defer cancel()
+			cl, err := l.EndpointProvider.EthClient(ctx)
+			if err != nil {
+				l.Log.Error("Can't reach sequencer execution RPC", "err", err)
+				continue
+			}
+			pendingBytes := l.state.PendingBytes()
+			var maxTxSize, maxBlockSize uint64
+			if pendingBytes > int64(l.Config.ThrottleThreshold) {
+				l.Log.Warn("Pending bytes over limit, throttling DA", "bytes", pendingBytes, "limit", l.Config.ThrottleThreshold)
+				maxTxSize = l.Config.ThrottleTxSize
+				maxBlockSize = l.Config.ThrottleBlockSize
+			}
+			var success bool
+			if err := cl.Client().CallContext(
+				ctx, &success, SetMaxDASizeMethod, hexutil.Uint64(maxTxSize), hexutil.Uint64(maxBlockSize)); err != nil {
+				l.Log.Error("SetMaxDASize rpc failed:", err)
+			}
+			if !success {
+				l.Log.Error("Result of SetMaxDASize was false")
+				continue
+			}
+			if l.Config.ThrottleThreshold == 0 {
+				// now that we've disabled any potential previous DA filter, since throttling is disabled, there's no
+				// need to continue looping.
```

Comment:
Hmm, @0x00101010 do you think we should just keep looping anyway? I wonder about having multiple sequencer HA instances, perhaps if we fail over to another one it might have had throttling active last time it was used.

Reply:
I think it is important that we loop here, if we ever need to change leader sequencer, then having this loop can help us set the parameters to the correct one (within the interval). Otherwise the throttling config we want won't be configured on the new leader

Reply:
yeah, better safe than sorry, not exactly a lot of overhead to do this anyway.  dropped the early exit.